[Request] [8.11.4 & 8.12][ESS] Remove mention of default query in the ES|QL tab docs (#4578)
elastic/security-docs: docs/events/timeline-ui-overview.asciidoc

You can use {esql} in Timeline by opening the **{esql}** tab. From there, you can: 

- Write an {esql} query to explore your events. For example, start with the following query, then iterate on it to tailor your results: 
+
[source,esql]
----
FROM .alerts-security.alerts-default,apm-*-transaction*,auditbeat-*,endgame-*,filebeat-*,logs-*,packetbeat-*,traces-apm*,winlogbeat-*,-*elastic-cloud-logs-*
| LIMIT 10
| KEEP @timestamp, message, event.category, event.action, host.name, source.ip, destination.ip, user.name
----
+
This query does the following:

** It starts by querying documents within the Security alert index (`.alerts-security.alerts-default`) and indices specified in the <<update-sec-indices,Security data view>>.
** Then, the query limits the output to the top 10 results.

> ** Next, it keeps the `@timestamp`, `message`, `event.category`, `event.action`, `host.name`, `source.ip`, `destination.ip`, and `user.name` fields in the output.

Author: ```suggestion
** Finally, it keeps the default `@timestamp`, `message`, `event.category`, `event.action`, `host.name`, `source.ip`, `destination.ip`, and `user.name` Timeline fields in the output.
```

> NOTE: An error message displays when the query bar is empty. 

Reviewer: Would we also note here? "Without the keep clause, all fields for every index that matches the given `FROM` pattern will be requested. For any indices with a significant number of fields such as indices under `logs-*` this could lead to decreased performance of the query.

Not sure if this is the best place for that, but we should highlight that for users somewhere I think

Author: I can expand the note to include these performance notes.

Author: Took an initial stab at incorporating this information in [dbbff14](https://github.com/elastic/security-docs/pull/4578/commits/dbbff141b5cb1a4ac50a0c30a1a13dfbcb440e92) and [bf27e88](https://github.com/elastic/security-docs/pull/4578/commits/bf27e88bc613eaf00396833dffdd6e872f7c23c9).

You can use {esql} in Timeline by opening the **{esql}** tab. From there, you can: 

- Write an {esql} query to explore your events. For example, start with the following query, then iterate on it to tailor your results: 
+
[source,esql]
----
FROM .alerts-security.alerts-default,apm-*-transaction*,auditbeat-*,endgame-*,filebeat-*,logs-*,packetbeat-*,traces-apm*,winlogbeat-*,-*elastic-cloud-logs-*
| LIMIT 10
| KEEP @timestamp, message, event.category, event.action, host.name, source.ip, destination.ip, user.name
----
+
This query does the following:

** It starts by querying documents within the Security alert index (`.alerts-security.alerts-default`) and indices specified in the <<update-sec-indices,Security data view>>.

> ** Then, the query limits the output to the top 10 results.

Reviewer 2: Would this mean the top 10 most recent documents? Would that be worth mentioning?

Author: @michaelolo24 do you remember what the Discover folks said about this? [Here's](https://github.com/elastic/security-docs/pull/4065#discussion_r1374985875) the convo we had in 8.11 on this topic.

Reviewer 2: For the record I did review the ES|QL [doc ](https://security-docs_bk_4578.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/esql-commands.html#esql-limit) about LIMIT and it doesn't seem to say.

Reviewer: Hey, yea. They don't actually guarantee it's the latest 10. They are randomly determined.

Author: Resolving this since we haven't doc'd this information in the past.

> ** Finally, it keeps the default `@timestamp`, `message`, `event.category`, `event.action`, `host.name`, `source.ip`, `destination.ip`, and `user.name` Timeline fields in the output.

Reviewer 2: ```suggestion
** Finally, it keeps the default `@timestamp`, `message`, `event.category`, `event.action`, `host.name`, `source.ip`, `destination.ip`, and `user.name` fields in the output.
```
Not sure it makes sense to call them Timeline fields instead of just fields

Author: While I think both versions work, I feel like the original is slightly more accurate and descriptive. Similar to the default fields that display on the rule details page, we also have default fields that display in Timeline (example below):

<img width="800" alt="Screenshot 2024-01-10 at 11 54 28 AM" src="https://github.com/elastic/security-docs/assets/79124755/a01ed052-8720-4abd-b9d4-d9a9f3c4da48">

If I moved "Timeline" closer to "default", would that be more clear?

```suggestion
** Finally, it keeps the default Timeline fields (`@timestamp`, `message`, `event.category`, `event.action`, `host.name`, `source.ip`, `destination.ip`, and `user.name`) in the output.
```

+

> TIP: When querying indices that tend to be large (for example, `logs-*`), performance can be impacted by the number of fields being returned in the output. To optimize performance, we recommend using the {ref}/esql-commands.html#esql-keep[`KEEP`] command to specify fields that you want returned. For example, add the clause `KEEP @timestamp, user.name` to the end of your query to specify that you only want the `@timestamp` and `user.name` fields returned.

Reviewer 2: ```suggestion
TIP: When querying indices that tend to be large (for example, `logs-*`), performance can be impacted by the number of fields returned in the output. To optimize performance, we recommend using the {ref}/esql-commands.html#esql-keep[`KEEP`] command to specify fields that you want returned. For example, add the clause `KEEP @timestamp, user.name` to the end of your query to specify that you only want the `@timestamp` and `user.name` fields returned.
```

> TIP: When querying large indices (for example, `logs-*`), performance can be impacted by the number of fields being returned in the output. If your query is taking a long time to complete, try using the {ref}/esql-commands.html#esql-keep[`KEEP`] command and specifying the fields you want returned. For example, add the clause `KEEP @timestamp, user.name` to the end of your query to specify that you only want the `@timestamp` and `user.name` fields returned.

Author: ```suggestion
TIP: When querying indices that tend to be large (for example, `logs-*`), performance can be impacted by the number of fields being returned in the output. If your query is taking a long time to complete, try using the {ref}/esql-commands.html#esql-keep[`KEEP`] command and specifying the fields you want returned. For example, add the clause `KEEP @timestamp, user.name` to the end of your query to specify that you only want the `@timestamp` and `user.name` fields returned.
```

Author: @PhilippeOberti would you mind reviewing this as well? Thank you!

Reviewer 3: the only thing that I would change is this part: `If your query is taking a long time to complete`

I feel like we want to try to avoid having customers running those query in the first place. So maybe phrase it a bit differently, like: `always try using the keep command unless necessary`?

Author: Ah, ok so frame using the `KEEP` clause as a best practice? How about this:

```suggestion
TIP: When querying indices that tend to be large (for example, `logs-*`), performance can be impacted by the number of fields being returned in the output. To optimize performance, use the {ref}/esql-commands.html#esql-keep[`KEEP`] command to specify the fields you want returned. For example, add the clause `KEEP @timestamp, user.name` to the end of your query to specify that you only want the `@timestamp` and `user.name` fields returned.
```

Author: Another option if "to optimize performance" is too large of a statement:

```suggestion
TIP: When querying indices that tend to be large (for example, `logs-*`), performance can be impacted by the number of fields being returned in the output. To reduce performance issues, use the {ref}/esql-commands.html#esql-keep[`KEEP`] command to specify fields you want returned. For example, add the clause `KEEP @timestamp, user.name` to the end of your query to specify that you only want the `@timestamp` and `user.name` fields returned.
```

Reviewer 3: I like the first one better, but the second one works as well. I would even add `To optimize performance, we highly recommend using the ...` or something like that.

Author: ```suggestion
TIP: When querying indices that tend to be large (for example, `logs-*`), performance can be impacted by the number of fields being returned in the output. To optimize performance, we recommend using the {ref}/esql-commands.html#esql-keep[`KEEP`] command to specify fields that you want returned. For example, add the clause `KEEP @timestamp, user.name` to the end of your query to specify that you only want the `@timestamp` and `user.name` fields returned.
```
